megago/megago.py

```diff
@@ -303,79 +151,37 @@ def __exit__(self, exc_type, exc_value, traceback):
         sys.stderr = self.old_stderr
 
 
-def run_process(ids, go1, go2, freq_dict, queue, godag=None):
-    if godag is None:
-        godag = GODag(GODAG_FILE_PATH, prt=LogFile())
-    for id in ids:
-        BMA_test = best_match_average(go1[id], go2[id], freq_dict, godag)
-        queue.put([id, BMA_test])
+def run_comparison(in_file):
+    """ Compute the pairwise similarity values for all rows from the given file.
+    Params
+    ------
+    in_file: The file for wich row-wise similarities should be computed.
+
+    Returns
+    -------
+    A string that represents a "CSV"-file with a similarity value per row.
+    """
```

Comment:
```suggestion
    """ Compute the pairwise similarity values for all rows from the given file.
    
    Parameters
    ------
    in_file : file_handle / path_to_file
        The file for which row-wise similarities should be computed.

    Returns
    -------
    str
        A string that represents a "CSV"-file with a similarity value per row.
    """
```

I am not sure about the type of the `in_file` parameter.